src/Symfony/Component/Serializer/Serializer.php

```diff
@@ -55,10 +56,14 @@ class Serializer implements SerializerInterface, ContextAwareNormalizerInterface
     protected $decoder;
 
     /**
-     * @var array
+     * @internal since Symfony 4.1
      */
     protected $normalizers = array();
 
+    private $cachedNormalizers;
```

Comment:
@nicolas-grekas what is this property for? it it set below but never used again.

Reply:
ah, `$this->normalizers` is protected so could be edited by extensions. in which case you'd want to reset the cache?

Reply:
yes, that's the purpose, allowing efficient caching without breaking BC of the protected property (which is not internal so will be removed in 5.0